frontend/src/components/MapRenderer.ts

```diff
@@ -16,20 +17,25 @@ export function updateDataSources(
   reports: any,
   relationships: any,
   box: Position[],
+  annotations: Annotations,
   map: mapboxgl.Map,
 ) {
   if (sourceLoaded) {
     const reportSource: mapboxgl.GeoJSONSource = map.getSource('reports') as mapboxgl.GeoJSONSource;
     const relationshipSource: mapboxgl.GeoJSONSource = map.getSource('relationships') as mapboxgl.GeoJSONSource;
     const boxSource: mapboxgl.GeoJSONSource = map.getSource('box') as mapboxgl.GeoJSONSource;
+    const pointAnnotationSource: mapboxgl.GeoJSONSource = map.getSource('points') as mapboxgl.GeoJSONSource;
```

Comment:
I know you're working on it so this is not a formal review, but as a small question, what happens if there is no source "points" in map?

Can that ever happen?

I think this might be the source of some map crashing bugs. We should have null safe checking when we call these sources if that is the case to avoid crashes.

Null safe in javascript can be done like:
```
nullableObject?.someFunction();
```
as opposed to:
```
if (nullableObject) nullableObject.someFunction();
```
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining